tests/expecter.go

```diff
@@ -201,9 +201,10 @@ func NewConsoleExpecter(virtCli kubecli.KubevirtClient, vmi *v1.VirtualMachineIn
 	}, timeout, opts...)
 }
 
-// ExpectBatchWithValidatedSend adds the expect.BSnd command to the exect.BExp expression.
+// ExpectBatchWithValidatedSend adds the expect.BSnd command to the expect.BExp expression.
 // It is done to make sure the match was found in the result of the expect.BSnd
 // command and not in a leftover that wasn't removed from the buffer.
+// It also expects the Bexp to be at the begining of the line with only CRLF expression preceding it.
```

Comment:
I don't understand the comment. How can `BExp` be in the beginning of the line? What line? Do you mean at the begining of the batch?

Anyway, I don't understand the commit message as well. Can you please provide an example?
Why the buffer go flushed?

I'm not sure since I didn't understand this commit.
But since I saw in the commit message the problem (which I don't understand:)) is mostly on login I just want to mention as I mentioned in the previous commit. `ExpectBatchWithValidatedSend` is not suitable for login methods.
The main idea behind `ExpectBatchWithValidatedSend` is that it makes sure the expected expression is the result of the sent command.
The method is not suitable for batches as the login where we first send `\n` but the expression we are looking for may appear in the buffer much before we sent the `\n`.

Reply:
I don't want to lose a bit of the safe expecter for two login methods. The logion methods are not suitable for `ExpectBatchWithValidatedSend` as I explained in the previous comment.